docs/authentication/keycloak.md

```diff
@@ -4,4 +4,238 @@ title: Authentication with Keycloak
 sidebar_label: Keycloak
 ---
 
-## TODO
\ No newline at end of file
+## Graphback Keycloak Authz
+
+Graphback Keycloak Authz enables [Keycloak](https://www.keycloak.org/) integration in [Graphback](https://graphback.dev) based applications. This enables you to declaratively add authorization capabilities like role based access on top of the CRUD model that is used within Graphback.
+
+This package is designed to work with [`keycloak-connect`](https://www.npmjs.com/package/keycloak-connect) and [`keycloak-connect-graphql`](https://www.npmjs.com/package/keycloak-connect-graphql). `keycloak-connect` is the official Keycloak middleware for Express applications. `keycloak-connect-graphql` provides deeper Keycloak integration into GraphQL servers.
+
+> NOTE: This package is an early alpha and not officially supported by Graphback
+
+## Getting Started
+
+This module requires you to install the following dependencies into your application.
+
+```bash
+npm install keycloak-connect
+npm install keycloak-connect-graphql
+```
+
+Then follow the [Getting started instructions for `keycloak-connect-graphql`](https://github.com/aerogear/keycloak-connect-graphql#getting-started)
+
+Once the getting started instructions are covered, you must create a configuration that defines the authorization rules for each model within your Graphback application.
+
+Here is an example configuration.
+
+```ts
+const authConfig = {
+  Task: {
+    create: {},
+    read: {},
+    update: { roles: ['admin'] },
+    delete: { roles: ['admin'] },
+    subCreate: {},
+    subUpdate: { roles: ['admin'] },
+    subDelete: { roles: ['admin'] },
+  },
+  Report: {
+    create: { roles: ['admin'] },
+    update: { roles: ['admin'] },
+    delete: { roles: ['admin'] },
+  },
+};
+```
+
+:::info
+`CrudServiceAuthConfig` interface defines all possible keys that each model can define.
```

Comment:
Once we have something like https://github.com/aerogear/graphback/issues/1702 we may be able to add link for the `CrudServiceAuthConfig` interface and all the other API.